### ISSUE TITLE
TPM2_ContextSave on public keys loaded using TPM2_LoadExternal leads to a TPM failure

### ISSUE THREAD

**Issue by diabonas** [stefanberger/libtpms#195]
**Describe the bug**
libtpms 0.8.0 since commit 1e1648fec31bd8d80b2ece4eaefe851e2cf4009d uses `ANY_OBJECT_Marshal` to save key contexts. However, public keys loaded using `TPM2_LoadExternal` have a [sensitive portion of `NULL`](https://github.com/stefanberger/libtpms/blob/921d6dc92e153722e3d186dbb92fc2be589e1c38/src/tpm2/ObjectCommands.c#L187), leading to the following error when trying to run `TPM2_ContextSave` on them:
```
libtpms/tpm2: Entering failure mode; code: 4, location: TPMU_SENSITIVE_COMPOSITE_Marshal line 2126
libtpms/tpm2: TPM2_Process: Entered failure mode through command:
80 01 00 00 00 0e 00 00 01 62 80 00 00 00 
```

**To Reproduce**
1. Install libtpms 0.8.0, swtpm 0.5.2, OpenSSL 1.1.1j and [tpm2-tools](https://github.com/tpm2-software/tpm2-tools/) 5.0.
2. Run the following commands:
    ```
    openssl genrsa -out priv.key 2048
    openssl rsa -in priv.key -pubout -out pub.key
    swtpm socket --tpm2 --server port=2321 --ctrl type=tcp,port=2322 --flags not-need-init,startup-clear --tpmstate dir=/tmp &
    tpm2_loadexternal --tcti=swtpm --key-algorithm=rsa --public=pub.key --key-context=key.ctx
    ```
3. Observe the last command failing with the error message
    ```
    libtpms/tpm2: Entering failure mode; code: 4, location: TPMU_SENSITIVE_COMPOSITE_Marshal line 2126
    libtpms/tpm2: TPM2_Process: Entered failure mode through command:
    80 01 00 00 00 0e 00 00 01 62 80 00 00 00
    WARNING:esys:src/tss2-esys/api/Esys_ContextSave.c:247:Esys_ContextSave_Finish() Received TPM Error
    ERROR:esys:src/tss2-esys/api/Esys_ContextSave.c:92:Esys_ContextSave() Esys Finish ErrorCode (0x00000101)
    ERROR: Esys_ContextSave(0x101) - tpm:error(2.0): commands not being accepted because of a TPM failure
    ERROR: Unable to run tpm2_loadexternal
    ```
4. With libtpms <= 0.7.5, the `tpm2_loadexternal` command succeeds, printing something like
    ```
    name: 000b5bf247d84aedc11dd556c59aa600386cb28de8e03be313c82ddd2177911bd8c5
    ```
    in return instead.

**Expected behavior**
Saving a context for a public key loaded using  `TPM2_LoadExternal` should work without producing an internal TPM failure.

**Desktop**
 - OS: Arch Linux (rolling release)

**Versions of relevant components**
- libtpms: 0.8.0
- openssl: 1.1.1j
- swtpm 0.5.2

**Additional context**
This breaks some tests of the tpm2-tools test suite which rely on externally loaded public keys.

**Comment by stefanberger** [comment]
Thank you so much for reporting this! I can recreate the issue with the IBM TSS2 tools as well:

Start swtpm in one terminal:
```
swtpm socket --tpmstate dir=/tmp/myvtpm --tpm2 --ctrl type=tcp,port=2322    --server type=tcp,port=2321 --flags not-need-init
```

Other terminal:
```
export TPM_COMMAND_PORT=2321 TPM_PLATFORM_PORT=2322 \
  TPM_SERVER_NAME=localhost TPM_INTERFACE_TYPE=socsim \
  TPM_SERVER_TYPE=raw
tssstartup
openssl genrsa -out priv.key 2048
openssl rsa -in priv.key -pubout -out pub.key
tssloadexternal -hi o -ipem pub.key
tsscontextsave -ha 80000000
```

I will post a patch soon.

**Comment by stefanberger** [comment]
@diabonas Can you test this with the patch in PR #196? I may add a test case in the meantime as well...

**Comment by diabonas** [comment]
This is fixed in version [0.8.1](https://github.com/stefanberger/libtpms/releases/tag/v0.8.1), so I am closing this issue - thanks again!
